akka-actor/src/main/scala/akka/util/ByteString.scala

```diff
@@ -528,7 +528,7 @@ object CompactByteString {
     else {
       val ar = new Array[Byte](bytes.size)
       bytes.copyToArray(ar)
-      CompactByteString(ar)
+      ByteString.ByteString1C(ar)
```

Comment:
I'd run some benchmarks on different implementations of `apply` and received following results:

First, when passing several byte arguments `toArray.clone` seems to be slightly faster than `copyToArray`:

```
VarargsCopy          thrpt  600   82420.414 ±  707.086  ops/ms
VarargsToArrayClone  thrpt  600   85950.390 ±  684.836  ops/ms
```

Also, there is an old implementation (DoubleCopy), and toArray without copy, just for comparison:

```
VarargsDoubleCopy    thrpt  600   61192.525 ±  542.109  ops/ms
VarargsToArray       thrpt  600  132025.408 ± 1357.725  ops/ms
```
But if passing varargs from some existing seq like `ByteString(someSeq: _*)` `toArray` call becomes very expensive:

```
SeqCopy              thrpt  600   24281.646 ±  223.725  ops/ms
SeqToArrayClone      thrpt  600    2720.174 ±   20.391  ops/ms
SeqDoubleCopy        thrpt  600   19321.256 ±  164.948  ops/ms
SeqToArray           thrpt  600    2835.829 ±   23.124  ops/ms
```

I think that it's because of reflection, which is used to create array in [`TraversableOnce.toArray`](https://github.com/scala/scala/blob/v2.11.8/src/library/scala/collection/TraversableOnce.scala#L285) from `ClassTag`.

So, if second scenario important enough? If it's not, then it really make sense to just use `toArray.clone`, and I'll update the PR.